### ISSUE TITLE
Hotloading doesn't work for subdirectories

### ISSUE THREAD

**Issue by A5308Y** [abevoelker/rails-elm-example#1]
Hello!

Thank you for the great tutorial and your example repo. It made setting up elm for rails really easy!

I'm having some trouble with live reloading though. I've split up my elm files in subdirectories. Now live reloading only happens if I change something in a file living directly in webpack/elm. elm files in subdirectories don't trigger a compilation if changed.

I looked at the webpack documentation and I guess I have to adjust `resolve` in webpack.config.js, but I'm confused by the documentation:

- Do I have to add another `root` entry?
- Should I use `moduleDirectories`
- Do I have to set watch options?
- Should I set `entry`?
- Do I have to do a combination of the above?
- ...

I tried around for a while but couldn't get anything to change. Can you help me? Maybe it might also be helpful to add an example for this to the repo.

Thanks in advance!

**Comment by abevoelker** [comment]
Hi, thanks for the feedback! Truthfully I switched off from using Elm shortly after I got it working. I loved Elm's type safety, but I didn't like the flash-of-no-content from not having server-side rendering, and it seemed that for some functionality I needed I would have to spend time writing small libraries myself rather than using something that already exists. So I switched to React.

So unfortunately I haven't encountered your issue and don't have a solution for you. Nor enough experience with Webpack to really help you out in resolving it (writing the tutorial was my first and only foray into Webpack).

If you or someone else is able to solve your problem, I'll definitely update the tutorial and/or add an example! I'll leave the issue open until then. Thanks!

**Comment by A5308Y** [comment]
Ok Thanks. I'll ask around and see if I can come up with a solution.

**Comment by A5308Y** [comment]
Hi again,

I finally found time to address my issue. As it turns out I didn't understand how name spacing works with folders. I named modules in folder Foo "module Bar exposing(..)" instead of "module Foo.Bar exposing(..)". This way I had to include all the source directories in elm-package.json, but I thought that's just the way elm imports work.

After name spacing all my modules reloading works for subdirectories as well with the setup from your blog post.
